association_backend/association/domain/services/handle_customer_subscription_updated.py

```diff
@@ -0,0 +1,52 @@
+import logging
+
+from pydantic import BaseModel
+
+from association.domain.entities.stripe import StripeStatus
+from association.domain.entities.subscriptions import SubscriptionState
+from association.domain.exceptions import (
+    InconsistentStateTransitionError,
+    SubscriptionNotFound,
+)
+from association.domain.repositories.association_repository import AssociationRepository
+
+logger = logging.getLogger(__name__)
+
+
+class SubscriptionDetailInput(BaseModel):
+    subscription_id: str
+    status: str
+
+
+async def handle_customer_subscription_updated(
+    data: SubscriptionDetailInput, association_repository: AssociationRepository
+):
+    subscription = await association_repository.get_subscription_by_stripe_id(
+        data.subscription_id
+    )
+    if subscription:
+        # if subscription.is_for_life:
+        #     subscription.state = SubscriptionState.ACTIVE
+        if data.status == StripeStatus.ACTIVE:
+            subscription.state = SubscriptionState.ACTIVE
+        elif data.status == StripeStatus.INCOMPLETE:
+            subscription.state = SubscriptionState.PENDING
+        elif data.status == StripeStatus.INCOMPLETE_EXPIRED:
+            if subscription.stripe_id:
+                error_message = (
```

Comment:
reminder for me, this check is wrong
I have to check if there are payments associated to the old subscription or if it is set as ACTIVE or EXPIRED